File: MDVision.IEHR/Controls/Admin/Admin_FeeGroup_Detail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MDVision.Datasets;
using MDVision.Business.BCommon;
using System.Data;

using Newtonsoft.Json;
using MDVision.Common.Utilities;
using MDVision.Common.Shared;
using MDVision.Business.BLL;
using MDVision.IEHR.Common;

namespace MDVision.IEHR.Controls.Admin
{
    public class Admin_FeeGroup_Detail
    {
        private BLLFeeSchedule BLLFeeScheduleObj = null;
        public Admin_FeeGroup_Detail()
        {
            BLLFeeScheduleObj = new BLLFeeSchedule();
        }
        #region Singleton
        private static Admin_FeeGroup_Detail _obj = null;
        public static Admin_FeeGroup_Detail Instance()
        {
            if (_obj == null)
                _obj = new Admin_FeeGroup_Detail();
            return _obj;
        }
        #endregion

        #region Private Functions
        /// <summary>
        /// Saves the Fee Group.
        /// </summary>
        /// <param name="fieldsJSON">The fields json.</param>
        /// <returns>Json string containing Succes message or Exception message</returns>
        private string SaveFeeGroup(string fieldsJSON)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Fee Group", "ADD")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    System.Web.Script.Serialization.JavaScriptSerializer ser = new System.Web.Script.Serialization.JavaScriptSerializer();
                    var SearchedfieldsJSON = ser.Deserialize<dynamic>(fieldsJSON);

                    DSFeeSchedule dsFeeGroup = new DSFeeSchedule();
                    DSFeeSchedule.FeeGroupRow dr = dsFeeGroup.FeeGroup.NewFeeGroupRow();

                    dr.ShortName = SearchedfieldsJSON["txtName"];
                    dr.Description = SearchedfieldsJSON["txtDescription"];
                    if (!string.IsNullOrEmpty(MDVUtility.ToStr(SearchedfieldsJSON["txtEndDate"])))
                        dr.EndDate = MDVUtility.ToDateTime(SearchedfieldsJSON["txtEndDate"]);
                    if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlNextFeeGroup"]))
                        dr.NextFeeGroupId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlNextFeeGroup"]);

                    dr.IsActive = MDVUtility.ToStr(SearchedfieldsJSON["chkIsActive"]) == "True" ? true : false;

                    dr.CreatedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                    dr.CreatedOn = DateTime.Now;
                    dr.ModifiedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                    dr.ModifiedOn = DateTime.Now;
                    if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlEntity"]))
                        dr.EntityId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlEntity"]);

                    #region Database Insertion
                    dsFeeGroup.FeeGroup.AddFeeGroupRow(dr);
                    BLObject<DSFeeSchedule> obj = BLLFeeScheduleObj.InsertFeeGroup(ref dsFeeGroup);
                    dsFeeGroup = obj.Data;
                    if (obj.Data != null)
                    {
                        var response = new
                        {
                            status = true,
                            message = Common.AppPrivileges.Save_Message,
                            FeeGroupId = dsFeeGroup.Tables[dsFeeGroup.FeeGroup.TableName].Rows[0][dsFeeGroup.FeeGroup.FeeGroupIdColumn.ColumnName]
                        };
                        return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = obj.Message
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    #endregion
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Updates the FeeGroup Link.
        /// </summary>
        /// <param name="fieldsJSON">The fields json.</param>
        /// <param name="ProcedureCategoryId">The FeeGroup identifier.</param>
        /// <returns>Json string containing Succes message or Exception message</returns>
        private string UpdateFeeGroup(string fieldsJSON, Int64 FeeGroupId)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Fee Group", "EDIT")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    System.Web.Script.Serialization.JavaScriptSerializer ser = new System.Web.Script.Serialization.JavaScriptSerializer();
                    var SearchedfieldsJSON = ser.Deserialize<dynamic>(fieldsJSON);

                    DSFeeSchedule dsFeeGroup = new DSFeeSchedule();
                    //DSFeeSchedule.FeeGroupRow dr = dsFeeGroup.FeeGroup.NewFeeGroupRow();
                    BLObject<DSFeeSchedule> objLoad = BLLFeeScheduleObj.LoadFeeGroup(FeeGroupId, null, null, null, null);
                    dsFeeGroup = objLoad.Data;
                    foreach (DSFeeSchedule.FeeGroupRow dr in dsFeeGroup.Tables[dsFeeGroup.FeeGroup.TableName].Rows)
                    {
                        //dr.FeeGroupId = FeeGroupId;
                        dr.ShortName = SearchedfieldsJSON["txtName"];
                        dr.Description = SearchedfieldsJSON["txtDescription"];
                        if (!string.IsNullOrEmpty(MDVUtility.ToStr(SearchedfieldsJSON["txtEndDate"])))
                            dr.EndDate = MDVUtility.ToDateTime(SearchedfieldsJSON["txtEndDate"]);
                        else
                            dr[dsFeeGroup.FeeGroup.EndDateColumn] = DBNull.Value;
                        if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlNextFeeGroup"]))
                            dr.NextFeeGroupId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlNextFeeGroup"]);
                        else
                            dr.NextFeeGroupId = MDVUtility.ToInt64(0);
                        dr.IsActive = MDVUtility.ToStr(SearchedfieldsJSON["chkIsActive"]) == "True" ? true : false;
                        //dr.CreatedBy = "";
                        //dr.CreatedOn = DateTime.Now;
                        dr.ModifiedBy = MDVUtility.DecryptFrom64(MDVSession.Current.AppUserName);
                        dr.ModifiedOn = DateTime.Now;
                        if (!string.IsNullOrEmpty(SearchedfieldsJSON["ddlEntity"]))
                            dr.EntityId = MDVUtility.ToInt64(SearchedfieldsJSON["ddlEntity"]);
                    }

                    #region Database Updation
                    //dsFeeGroup.FeeGroup.AddFeeGroupRow(dr);
                    //dsFeeGroup.FeeGroup.AcceptChanges();

                    if (dsFeeGroup.Tables[dsFeeGroup.FeeGroup.TableName].Rows.Count > 0)
                    {
                        //dsFeeGroup.FeeGroup.Rows[0].SetModified();
                        BLObject<DSFeeSchedule> obj = BLLFeeScheduleObj.UpdateFeeGroup(ref dsFeeGroup);
                        if (obj.Data != null)
                        {
                            var response = new
                            {
                                status = true,
                                message = Common.AppPrivileges.Update_Message
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = obj.Message
                            };
                            return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                        }
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = ""
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    #endregion
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Fills the FeeGroup.
        /// </summary>
        /// <param name="ProcedureCategoryId">The Plan Fee Link identifier.</param>
        /// <returns>Json string containing key value pair or Exception message</returns>
        private string FillFeeGroup(Int64 FeeGroupId)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Fee Group", "VIEW")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    if (string.IsNullOrEmpty(MDVUtility.ToStr(FeeGroupId)))
                    {
                        var response = new
                        {
                            status = false,
                            Message = MDVUtility.ToStr(Common.AppPrivileges.Select_Message)
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    else
                    {
                        DSFeeSchedule dsFeeGroup = null;
                        BLObject<DSFeeSchedule> obj = BLLFeeScheduleObj.LoadFeeGroup(FeeGroupId, null, null, null, null);
                        if (obj.Data != null)
                        {
                            dsFeeGroup = obj.Data;
                            if (dsFeeGroup.Tables[dsFeeGroup.FeeGroup.TableName].Rows.Count > 0)
                            {
                                DataRow dr = dsFeeGroup.Tables[dsFeeGroup.FeeGroup.TableName].Rows[0];
                                var keyValues = new Dictionary<string, string>
                        {
                          { "txtName", MDVUtility.ToStr(dr[dsFeeGroup.FeeGroup.ShortNameColumn.ColumnName])},
                          { "txtDescription", MDVUtility.ToStr(dr[dsFeeGroup.FeeGroup.DescriptionColumn.ColumnName])},
                          //{ "txtEndDate", MDVUtility.ToStr(dr[dsFeeGroup.FeeGroup.EndDateColumn.ColumnName])},
                          { "txtEndDate", MDVUtility.ToStr(dr[dsFeeGroup.FeeGroup.EndDateColumn.ColumnName]) !="" ? MDVUtility.ToDateTime(dr[dsFeeGroup.FeeGroup.EndDateColumn.ColumnName]).ToShortDateString():""},
                          { "ddlNextFeeGroup", MDVUtility.ToStr(dr[dsFeeGroup.FeeGroup.NextFeeGroupIdColumn.ColumnName])},
                          { "ChkIsActive", MDVUtility.ToStr(dr[dsFeeGroup.FeeGroup.IsActiveColumn.ColumnName])},
                          { "ddlEntity", MDVUtility.ToStr(dr[dsFeeGroup.FeeGroup.EntityIdColumn.ColumnName])}
                        };
                                System.Web.Script.Serialization.JavaScriptSerializer js = new System.Web.Script.Serialization.JavaScriptSerializer();
                                var response = new
                                {
                                    status = true,
                                    FeeGroupFill_JSON = js.Serialize(keyValues)
                                };
                                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                            }
                            else
                            {
                                var response = new
                                {
                                    status = false,
                                    Message = Common.AppPrivileges.No_Record_Message,
                                };
                                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                            }
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = obj.Message
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }

                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }


        /// <summary>
        /// Fills the next fee group.
        /// </summary>
        /// <param name="EntityID">The entity identifier.</param>
        /// <returns>System.String.</returns>
        private string FillNextFeeGroup(string EntityID)
        {
            try
            {
                DSFeeSchedule dsFeeGroup = null;
                BLObject<DSFeeSchedule> obj = BLLFeeScheduleObj.LoadFeeGroup(0, null, null, EntityID, null);
                dsFeeGroup = obj.Data;
                var response = new
                {
                    status = true,
                    NextFeeGroupCount = dsFeeGroup.Tables[dsFeeGroup.FeeGroup.TableName].Rows.Count,
                    NextFeeGroupLoad_JSON = MDVUtility.JSON_DataTable(dsFeeGroup.Tables[dsFeeGroup.FeeGroup.TableName]),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        /// <summary>
        /// Deletes the FeeGroup.
        /// </summary>
        /// <param name="ProcedureCategoryId">The Plan Fee Link identifier.</param>
        /// <returns>Json string containing Succes message or Exception message</returns>
        private string DeleteFeeGroup(Int64 FeeGroupId)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Fee Group", "DELETE")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    if (string.IsNullOrEmpty(MDVUtility.ToStr(FeeGroupId)))
                    {
                        var response = new
                        {
                            status = false,
                            Message = MDVUtility.ToStr(Common.AppPrivileges.CheckBox_Message)
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                    else
                    {
                        BLObject<string> obj = BLLFeeScheduleObj.DeleteFeeGroup(MDVUtility.ToStr(FeeGroupId));
                        if (obj.Data == "")
                        {
                            var response = new
                            {
                                status = true,
                                Message = Common.AppPrivileges.Delete_Message
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = obj.Data
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }

        private string UpdateFeeGroupIsActive(Int64 FeeGroupId, Int64 IsActive)
        {
            try
            {
                string privilegesMessage = JsonConvert.DeserializeObject(AppPrivileges.GetFormSecurity("Fee Group", "EDIT")).ToString();
                if (string.IsNullOrEmpty(privilegesMessage))
                {
                    DSFeeSchedule dsFeeGroup = null;
                    BLObject<DSFeeSchedule> obj = BLLFeeScheduleObj.LoadFeeGroup(FeeGroupId, null, null, null, null);
                    dsFeeGroup = obj.Data;
                    if (dsFeeGroup.Tables[dsFeeGroup.FeeGroup.TableName].Rows.Count > 0)
                    {
                        DataRow dr = dsFeeGroup.Tables[dsFeeGroup.FeeGroup.TableName].Rows[0];
                        dr[dsFeeGroup.FeeGroup.IsActiveColumn.ColumnName] = IsActive;

                        BLObject<DSFeeSchedule> objUser = BLLFeeScheduleObj.UpdateFeeGroup(ref dsFeeGroup);
                        string successMsg;
                        if (objUser.Data != null)
                        {
                            if (IsActive == 0)
                                successMsg = Common.AppPrivileges.Inactive_Message;
                            else
                                successMsg = Common.AppPrivileges.Active_Message;
                            var response = new
                            {
                                status = true,
                                message = successMsg
                            };
                            return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
                        }
                        else
                        {
                            var response = new
                            {
                                status = false,
                                Message = objUser.Message
                            };
                            return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                        }
                    }
                    else
                    {
                        var response = new
                        {
                            status = false,
                            Message = obj.Message
                        };
                        return Newtonsoft.Json.JsonConvert.SerializeObject(response);
                    }
                }
                else
                {
                    var response = new
                    {
                        status = false,
                        Message = privilegesMessage
                    };
                    return (JsonConvert.SerializeObject(response));
                }
            }
            catch (Exception ex)
            {
                var response = new
                {
                    status = false,
                    Message = MDVCustomException.HumanReadableMessage(ex.Message),
                };
                return (Newtonsoft.Json.JsonConvert.SerializeObject(response));
            }
        }
        #endregion

        #region Service Command Handler
        /// <summary>
        /// Handle the Plan Fee Link Detail Commands and call to the respective methods.
        /// </summary>
        /// <param name="context">The context.</param>
        public void CommandHandler(HttpContext context)
        {
            string cammandAction = context.Request.QueryString["cammandAction"].ToUpper();

            switch (cammandAction)
            {
                case "SAVE_FEE_GROUP":
                    {
                        string fieldsJSON = context.Request["FeeGroupData"];
                        string strJSONData = SaveFeeGroup(fieldsJSON);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "FILL_FEE_GROUP":
                    {
                        string strFeeGroupId = context.Request["FeeGroupID"];
                        string strJSONData = FillFeeGroup(MDVUtility.ToInt64(strFeeGroupId));

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "FILL_NEXT_FEE_GROUP":
                    {
                        string strEntityId = context.Request["EntityID"];
                        string strJSONData = FillNextFeeGroup(strEntityId);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "DELETE_FEE_GROUP":
                    {
                        string strFeeGroupId = context.Request["FeeGroupID"];
                        string strJSONData = DeleteFeeGroup(MDVUtility.ToInt64(strFeeGroupId));

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "UPDATE_FEE_GROUP":
                    {
                        string fieldsJSON = context.Request["FeeGroupData"];
                        Int64 FeeGroupID = MDVUtility.ToInt64(context.Request["FeeGroupID"]);
                        string strJSONData = UpdateFeeGroup(fieldsJSON, FeeGroupID);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
                case "UPDATE_FEE_GROUP_ACTIVE_INACTIVE":
                    {
                        Int64 FeeGroupID = MDVUtility.ToInt64(context.Request["FeeGroupID"]);
                        Int64 IsActive = MDVUtility.ToInt64(context.Request["IsActive"]);
                        string strJSONData = UpdateFeeGroupIsActive(FeeGroupID, IsActive);

                        context.Response.ContentType = "text/plain";
                        context.Response.Write(strJSONData);
                    }
                    break;
            }
        }
        #endregion
    }
}

File: MDVision.IEHR/Controls/Admin/Admin_FeeGroup_Detail.js
feegroupDetail = {
    params: [],
    Enable: false,
    Load: function (params) {
        feegroupDetail.params = params;

        var self = null;
        if (feegroupDetail.params.PanelID == "feegroupDetail")
            self = $('#feegroupDetail');
        else
            self = $('#' + feegroupDetail.params.PanelID + ' #feegroupDetail');

        if (globalAppdata.AppUserName.toLowerCase() != "mdvision") {
            self.find("#ddlEntity").attr('disabled', 'disabled');
        }
        self.loadDropDowns(true).done(function () {            
            //if (globalAppdata['IsAdmin'] != "True") {
            //    $("#feegroupDetail #divFeeGroup_Entity").css("display", "none");
            //    $("#feegroupDetail #ddlEntity").val(globalAppdata["SeletedEntityId"]);
            //}
            if (globalAppdata.AppUserName.toLowerCase() != "mdvision") {
                self.find("#ddlEntity").val(globalAppdata["SeletedEntityId"]);
            }

            $("#feegroupDetail #chkIsActive").attr("checked", true);
            feegroupDetail.LoadFeeGroup();
           
        });
           
    },


    LoadFeeGroup: function () {
        if (feegroupDetail.params.mode == "Add") {

            //serialize data
            $('#frmFeeGroupDetail').data('serialize', $('#frmFeeGroupDetail').serialize());
            feegroupDetail.ValidationFeeGroup();
        }
        else if (feegroupDetail.params.mode == "Edit") {
            feegroupDetail.FillFeeGroup(feegroupDetail.params.FeeGroupId).done(function (response) {
                if (response.status != false) {
                    var feegroup_detail = JSON.parse(response.FeeGroupFill_JSON);
                    var self = $("#feegroupDetail");

                    utility.bindMyJSON(true, feegroup_detail, false, self).done(function () {

                        if (feegroup_detail.ChkIsActive == 'True') {
                            $("#feegroupDetail #chkIsActive").attr("checked", true);
                        }
                        else {
                            $("#feegroupDetail #chkIsActive").attr("checked", false);
                        }
                        feegroupDetail.LoadNextFeeGroup(feegroup_detail.ddlEntity, feegroup_detail.ddlNextFeeGroup);
                        feegroupDetail.ValidationFeeGroup();

                        //serialize data
                        $('#frmFeeGroupDetail').data('serialize', $('#frmFeeGroupDetail').serialize());

                    });

                   
                }
                else {
                    utility.DisplayMessages(response.Message, 3);
                }
            });
        }
    },

    LoadNextFeeGroup: function (entityID, ddlNextFeeGroup) {
        var EntityID = "";
        feegroupDetail.FillNextFeeGroup(entityID).done(function (response) {
            if (response.status != false) {
                var feegroup_detail = JSON.parse(response.NextFeeGroupLoad_JSON);
                $("#tblfeegroupDetail #ddlNextFeeGroup").empty();
                $("#tblfeegroupDetail #ddlNextFeeGroup").append($('<option/>', {
                    value: "",
                    html: "- SELECT -"
                }));
                $.each(feegroup_detail, function (i, item) {
                    $("#tblfeegroupDetail #ddlNextFeeGroup").append(
                        $('<option/>', {
                            value: item.FeeGroupId,
                            html: item.EntityName + " - " + item.ShortName
                        })
                    );
                });
            }
            $('#tblfeegroupDetail #ddlNextFeeGroup').val(ddlNextFeeGroup);
        });
    },

    FeeGroupSave: function () {
        var strMessage = "";
        var self = $("#feegroupDetail");
        var myJSON = self.getMyJSON();
        if (feegroupDetail.params.mode == "Add") {
            AppPrivileges.GetFormPrivileges("Fee Group", "ADD", "FORM_PRIVILEGE", "GET_FORM_PRIVILEGE", function (strMessage) {
                if (strMessage == "") {
                    feegroupDetail.SaveFeeGroup(myJSON).done(function (response) {
                        if (response.status != false) {
                            Admin_FeeGroup.FeeGroupSearch(response.FeeGroupId);
                            utility.DisplayMessages(response.message, 1);
                            UnloadActionPan();
                            CacheManager.BindCodes('GetFeeGroup', true);
                        }
                        else {
                            utility.DisplayMessages(response.Message, 3);
                        }
                    });
                }
                else
                    utility.DisplayMessages(strMessage, 2);
            });
        }
        else if (feegroupDetail.params.mode == "Edit") {
            AppPrivileges.GetFormPrivileges("Fee Group", "EDIT", "FORM_PRIVILEGE", "GET_FORM_PRIVILEGE", function (strMessage) {
                if (strMessage == "") {
                    feegroupDetail.UpdateFeeGroup(myJSON, feegroupDetail.params.FeeGroupId).done(function (response) {
                        if (response.status != false) {
                            Admin_FeeGroup.FeeGroupSearch(feegroupDetail.params.FeeGroupId);

                            utility.DisplayMessages(response.message, 1);
                            UnloadActionPan();
                            CacheManager.BindCodes('GetFeeGroup', true);
                        }
                        else {
                            utility.DisplayMessages(response.Message, 3);
                        }
                    });
                }
                else
                    utility.DisplayMessages(strMessage, 2);
            });
        }
    },

    ValidationFeeGroup: function () {
        $('#frmFeeGroupDetail')
          .bootstrapValidator({
              live: 'disabled',
              message: 'This value is not valid',
              feedbackIcons: {
                  valid: 'glyphicon glyphicon-ok',
                  invalid: 'glyphicon glyphicon-remove',
                  validating: 'glyphicon glyphicon-refresh'
              },
              fields: {
                  Name: {
                      group: '.col-md-4',
                      validators: {
                          notEmpty: {
                              message: ''
                          },
                      }
                  },
                  Entity: {
                      group: '.col-md-4',
                      validators: {
                          notEmpty: {
                              message: ''
                          },
                      }
                  },
              }
          })
       .on('success.form.bv', function (e) {
           e.preventDefault();
           feegroupDetail.FeeGroupSave();
       });
    },

    SaveFeeGroup: function (FeeGroupData) {
        var data = "FeeGroupData=" + FeeGroupData;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_FEE_GROUP_DETAIL", "SAVE_FEE_GROUP");
    },

    UpdateFeeGroup: function (FeeGroupData, FeeGroupID) {
        var data = "FeeGroupData=" + FeeGroupData + "&FeeGroupID=" + FeeGroupID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_FEE_GROUP_DETAIL", "UPDATE_FEE_GROUP");
    },

    FillFeeGroup: function (FeeGroupID) {
        var data = "FeeGroupID=" + FeeGroupID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_FEE_GROUP_DETAIL", "FILL_FEE_GROUP");
    },

    FillNextFeeGroup: function (EntityID) {
        var data = "EntityID=" + EntityID;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_FEE_GROUP_DETAIL", "FILL_NEXT_FEE_GROUP");
    },

    UpdateFeeGroupActiveInactive: function (FeeGroupID, IsActive) {
        var data = "FeeGroupID=" + FeeGroupID + "&IsActive=" + IsActive;
        // serach parameter , class name, command name of class 
        return MDVisionService.defaultService(data, "ADMIN_FEE_GROUP_DETAIL", "UPDATE_FEE_GROUP_ACTIVE_INACTIVE");
    },

    UnLoad: function () {

        utility.UnLoadDialog("frmFeeGroupDetail", function () {
            UnloadActionPan();
        }, function () {
            UnloadActionPan();
        });

    },

    ShowHistory: function () {
        var PanelID = 'feegroupDetail';
        var ParentCtrl = 'feegroupDetail';
        var ProfileName = 'Fee Group';
        var DBTableName = 'FeeGroup';
        var ColumnKeyId = feegroupDetail.params.FeeGroupId

        utility.ShowHistory(ParentCtrl, PanelID, ProfileName, DBTableName, ColumnKeyId);
    },
}